src/test/java/com/radixdlt/common/EUIDTest.java

```diff
@@ -39,6 +46,72 @@
 		Arrays.fill(NEGATIVE_ONE, (byte) 0xff);
 	}
 
+	@Test
+	public void equalsContract() {
+		EqualsVerifier.forClass(EUID.class).verify();
+	}
+
+	@Test
+	public void testOffsetBytesConstructor() {
+		EUID expected = new EUID("dead000000000000000000000000beef");
+		byte[] tooManyBytes = Bytes.fromHexString("11dead000000000000000000000000beef");
+		EUID offsetted = new EUID(tooManyBytes, 1); // remember 2 hex chars == 1 byte.
+		assertEquals(expected, offsetted);
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void verify_that_exception_is_thrown_when_calling_constructor_with_too_short_hexstring() {
+		new EUID("deadbeef");
+		fail();
+	}
+
+	@Test(expected = IllegalArgumentException.class)
+	public void verify_that_exception_is_thrown_when_calling_constructor_with_empty_byte_array() {
+		new EUID(new byte[0]);
+		fail();
+	}
+
+	@Test
+	public void testZero() {
+		assertTrue(EUID.ZERO.isZero());
+		assertFalse(EUID.ONE.isZero());
+	}
+
+	@Test
+	public void testGetValue() {
+		assertEquals(EUID.ZERO.getValue(), UInt128.ZERO);
```

Comment:
I think the argument order here is reversed.  Should be `assertEquals(expectedValue, actualValue)`.

Reply:
Actually, all of these three here.

Reply:
yes, got it!